.github/workflows/create-dev-branch.yaml

```diff
@@ -0,0 +1,142 @@
+name: "Create Development Branch"
+run-name: Create a new development branch
+
+on:
+  workflow_dispatch:
+    inputs:
+      bump-type:
+        description: "bump type"
+        required: true
+        type: choice
+        default: "major"
+        options:
+          - "major"
+          - "minor"
+
+jobs:
+  create-branch:
+    runs-on: ubuntu-24.04
+    steps:
+      # validate branch
+      - name: Validate running branch
+        run: |
+          # we want to stop if this is not running on development branch
+          [[ "${{github.ref_name}}" =~ ^development/[0-9.]+$ ]] || exit 1
+      # create app token
+      - uses: actions/create-github-app-token@v1
+        id: app-token
+        with:
+          app-id: ${{ vars.ACTIONS_APP_ID }}
+          private-key: ${{ secrets.ACTIONS_APP_PRIVATE_KEY }}
+      # checkout
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ steps.app-token.outputs.token }}
+      # get version
+      - name: Get current version
+        run: |
+          source VERSION
+
+          SHORT_VERSION="$VERSION_MAJOR.$VERSION_MINOR"
+
+          echo "SHORT_VERSION=$SHORT_VERSION" >> $GITHUB_ENV
+      # calculate new version
+      - name: Calculate new version
+        run: |
+          source VERSION
+
+          if [[ "${{ inputs.bump-type }}" == "major" ]]; then
+            NEW_VERSION_MAJOR=$(($VERSION_MAJOR+1))
+            NEW_VERSION_MINOR=0
+          else
+            NEW_VERSION_MAJOR=$VERSION_MAJOR
+            NEW_VERSION_MINOR=$(($VERSION_MINOR+1))
+          fi
+
+          NEW_SHORT_VERSION="$NEW_VERSION_MAJOR.$NEW_VERSION_MINOR"
+
+          echo "NEW_SHORT_VERSION=$NEW_SHORT_VERSION" >> $GITHUB_ENV
+          echo "NEW_VERSION_MAJOR=$NEW_VERSION_MAJOR" >> $GITHUB_ENV
+          echo "NEW_VERSION_MINOR=$NEW_VERSION_MINOR" >> $GITHUB_ENV
+      # calculate new branch name
+      - name: Calculate new branch name
+        run: |
+          NEW_BRANCH_NAME="development/${{ env.NEW_SHORT_VERSION }}"
+          INIT_BRANCH_NAME="feature/init-dev-${{ env.NEW_SHORT_VERSION }}"
+
+          echo "NEW_BRANCH_NAME=$NEW_BRANCH_NAME" >> $GITHUB_ENV
+          echo "INIT_BRANCH_NAME=$INIT_BRANCH_NAME" >> $GITHUB_ENV
+      # create development branch
```

Comment:
Let's ensure the NEW_BRANCH does not exist already (like there is `development/129.0` you try to trigger create new major from `development/128.0` (I know that the push (or checkout -b) should fail but it's better to have a proper error message and no try to push to see that we have an issue)

Reply:
done in next step